pyspecProcScripts/integral_w_error.py

```diff
@@ -72,46 +72,18 @@ def integral_w_errors(
             "You have extra (non-phase cycling, non-indirect) dimensions: "
             + str(extra_dims)
         )
-    collected_variance = psp.ndshape(
-        [psp.ndshape(s)[indirect], len(error_path)], [indirect, "pathways"]
-    ).alloc()
-    avg_error = []
-    for j in range(len(error_path)):
-        # calculate N₂ Δf² σ², which is the variance of the integral
-        # (by error propagation) where N₂ is the number of points in
-        # the indirect dimension
-        s_forerror = select_pathway(s, error_path[j])
-        # previous line wipes everything out and starts over -- why not use
-        # collected_variance above, as I had originally set up --> part of
-        # issue #44
-        if j == 0:
-            N2 = psp.ndshape(s_forerror)[direct]
-        # mean divides by N₁ (indirect), integrate multiplies by Δf, and the
-        # mean sums all elements (there are N₁N₂ elements)
-        s_forerror -= s_forerror.C.mean_all_but([indirect, direct]).mean(
-            direct
-        )
-        s_forerror.run(lambda x: abs(x) ** 2 / 2).mean_all_but(
-            [direct, indirect]
-        ).mean(direct)
-        s_forerror *= df**2  # Δf
-        s_forerror *= N2
-        avg_error.append(s_forerror)
-    avg_error = sum(avg_error) / len(avg_error)
-    # {{{ variance calculation for debug
-    variance_debug = False
-    if variance_debug:
-        print(
-            "(inside automatic routine) the stdev seems to be",
-            np.sqrt(collected_variance / (df * N2)),
-        )
-        print(
-            "automatically calculated integral error:",
-            np.sqrt(collected_variance.data),
-        )
-    # }}}
-    s = select_pathway(s, sig_path)
-    retval = s.integrate(direct).set_error(psp.sqrt(s_forerror.data))
+    variance = calc_error(
+        s,
+        frq_slice,
+        sig_path,
+        indirect=indirect,
+        error_path=error_path,
+        use_mask=use_mask,
+        fl=fl,
+    )
+    retval = (
+        select_pathway(s, sig_path).integrate(direct).set_error(variance.data)
```

Comment:
the error should not be  set to the variance -- it should be set to the square root -- not sure  if this is a naming issue or what.

Reply:
OK, so having read more, I can clarify that your `calc_masked_variance` (as I'm suggesting you rename the function) will return *σ_ν²* (which is the **variance of your spectral datapoint** or, if you've taken a unitary FT, the **variance of your time domain datapoints**).  You then multiply by the other stuff (*Δν²* etc) and take the square root to determine the **estimated standard deviation of your integral,** which is what you attach as the error here.

Reply:
okay so now the calc_masked_variance only returns sigma_f^2